native/core/src/execution/datafusion/shuffle_writer.rs

```diff
@@ -139,6 +140,14 @@ impl ExecutionPlan for ShuffleWriterExec {
     ) -> Result<SendableRecordBatchStream> {
         let input = self.input.execute(partition, Arc::clone(&context))?;
         let metrics = ShuffleRepartitionerMetrics::new(&self.metrics, 0);
+        let read_time = MetricBuilder::new(&self.metrics).subset_time("read_time", 0);
+
+        // clone the ScanExec elapsed_compute metric
+        let scan_time = self
```

Comment:
Or we could just measure the time to execute the input?

Reply:
The input is a JNI call to fetch a batch that already exists, because CometExecIterator only invokes ShuffleWriterExec once the input batch is already available.

Reply:
Actually maybe you are correct. I will take another look today.